### PR TITLE
 added validation and test

### PR DESCRIPTION
**Pull request by ifzyy**
In this pull request:

- [x] Add the validations for the User Model
- [x]  Add the validations for the Post Model
- [x] Add Unit test case for the validations and method for all Models User, Post, Comment, Like

### CODE REVIEW

File: spec/models/post_spec.rb [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+  user = User.create(name: 'Harry', photo: 'https://unsplash.com/photos/pHPzdEHN6Os', bio: 'Web Designer')
+  describe 'Validation' do
+    it 'Post should be valid' do
+      post = Post.new(author: user, title: 'Full Stack Tutorial', text: 'This is complete tutorial',
+                      comments_counter: 3, likes_counter: 2)
+      expect(post).to be_valid
+    end
+
+    it 'Post should not be valid' do
+      Post.new(author: user, title: 'Full Stack Tutorial', text: 'This is complete tutorial',
```
Comment (Rustamxon7) [inline spec/models/post_spec.rb:13]: - [ ] You also need to write test for checking the length of your post title.


> Requirement: Title must not exceed 250 characters.

File: spec/models/post_spec.rb [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+    before { user.posts_counter = 3 }
+    it 'should increment the post counter' do
+      post = Post.new(author: user, title: 'Full Stack Tutorial', text: 'This is complete tutorial',
+                      comments_counter: 3, likes_counter: 2)
+      post.update_posts_counter
+      expect(user.posts_counter).to eq(4)
+    end
+  end
+
+  describe 'Method recent comment should return latest 5 comments' do
```
Comment (Rustamxon7) [inline spec/models/post_spec.rb:29]: - [ ] Write tests for these requirements as well. 


> 1. PostsCounter must be an `integer` `greater than or equal to zero`.
> 2. CommentsCounter must be an `integer` `greater than or equal to zero`.
> 3. LikesCounter must be an `integer` `greater than or equal to zero`.

**Review by Rustamxon7 — CHANGES_REQUESTED** [review]
Hello @ifzyy, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Highlights
_Your project is almost complete!_
_No Linters errors_
_Great Design_


## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better._

You can also consider:
N/A

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by zainsadaqat — APPROVED** [review]
# APPROVED!!!! 🎉 🎉 🎉 🎉 🎉 🎉
## Hey @ifzyy  😊 😃 

#### Great Job 😄 👏 You've done it. Your project is complete!  There is nothing else to say other than... it's time to merge it 👍🏻 ✅ 

# Congratulations!!! 🎉 🥳 👏 🙌 🎊 



Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**